wiki/DEFINITION_OF_DONE_CRITERIA.md

```diff
@@ -0,0 +1,100 @@
+# Defining cuML's Definition of Done Criteria
+
+
+## Algorithm Completion Checklist
+
+Below is a quick and simple checklist for developers to determine whether an algorithm is complete and ready for release. Most of these items contain more detailed descriptions in their corresponding developer guide. The checklist is broken down by layer (C++ or Python) and categorized further into
+
+- **Design:** All algorithms should be designed with an eye on maintainability, performance, readability, and robustness.
+- **Testing:** The goal for automated testing is to increase both the spread and the depth of code coverage as much as possible in order to ease time spent fixing bugs and developing new features. Additionally, a very important factor for a tool like `cuml` is to provide testing with multiple datasets that really stress the mathematical behavior of the algorithms. A comprehensive set of tests lowers the possibility for regressions and the introduction of bugs as the code evolves between versions. This covers both correctness & performance. 
+- **Documentation:** User-facing documentation should be complete and descriptive. Developer-facing documentation should be used for constructs which are complex and/or not immediately obvious. 
+
+### C++
+
+#### Design
+
+- Existing prims are used wherever possible
+- Array inputs and outputs to algorithms are accepted on device
+- New prims created wherever there is potential for reuse across different algorithms or prims
+- User-facing API is [stateless](cpp/DEVELOPER_GUIDE.md#public-cuml-interface) and follows the [plain-old data (POD)](https://en.wikipedia.org/wiki/Passive_data_structure) design paradigm
+- Public API contains a C-Wrapper around the stateless API
+- Public API contains an Scikit-learn-like stateful wrapper around the stateless API
+
+#### Testing
+
+- Prims: GTests with different inputs
+- Algorithms: End-to-end GTests with different inputs and different datasets
+
+#### Documentation
+
+- Complete and comprehensive Doxygen strings explaining the public API, restrictions, and gotchas. Any array parameters should also note whether the underlying memory is host or device.
+- Array inputs/outputs should also mention their expected size/dimension.
+- If there are references to the underlying algorithm, they must be cited too.
+
+
+### Python
+
+#### Design
+
+- Python class is as "near drop-in replacement" for Scikit-learn (or relevant industry standard) API as possible
```

Comment:
Expand a little - "parameters have the same names as scikit-learn, and where differences exist they are clearly documented in docstrings."